packages/contracts-bedrock/src/L1/OptimismPortalInterop.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+import { L1BlockInterop, ConfigType } from "src/L2/L1BlockInterop.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { Constants } from "src/libraries/Constants.sol";
+
+/// @custom:proxied
+/// @title OptimismPortalInterop
+/// @notice The OptimismPortal is a low-level contract responsible for passing messages between L1
+///         and L2. Messages sent directly to the OptimismPortal have no form of replayability.
+///         Users are encouraged to use the L1CrossDomainMessenger for a higher-level interface.
+contract OptimismPortalInterop is OptimismPortal {
+    /// @notice Thrown when a non-depositor account attempts update static configuration.
+    error Unauthorized();
+
+    /// @notice Reverts when the caller is not the SystemConfig contract.
+    modifier onlySystemConfig() {
+        if (msg.sender != address(systemConfig)) revert Unauthorized();
+        _;
+    }
+
+    /// @notice Constructs the OptimismPortal contract.
+    constructor() OptimismPortal() { }
+
+    /// @custom:semver 2.8.0+interop
+    function version() public pure override returns (string memory) {
+        return string.concat(super.version(), "+interop");
+    }
+
+    /// @notice Sets the gas paying token for the L2 system. This token is used as the
+    ///         L2 native asset. Only the SystemConfig contract can call this function.
+    /// @param _token    Address of the gas paying token.
+    /// @param _decimals Decimals of the gas paying token.
+    /// @param _name     Name of the gas paying token.
+    /// @param _symbol   Symbol of the gas paying token.
+    function setGasPayingToken(
+        address _token,
+        uint8 _decimals,
+        bytes32 _name,
+        bytes32 _symbol
+    )
+        external
+        override
+        onlySystemConfig
+    {
+        _setConfig(ConfigType.GAS_PAYING_TOKEN, abi.encode(_token, _decimals, _name, _symbol));
+    }
+
+    /// @notice Adds a chain to the interop dependency set.
```

Comment:
Can we update the portal such that it only has 1 generic function `setConfig` so we don't have to expand the ABI in multiple places? Right now we would have to make a diff to the portal and to the sys config every time there is a new config type, we should make it such that the portal just checks `msg.sender == systemConfig` then passes thru the data as a deposit

Reply:
This will allow us to add new types without needing to modify the portal, creating less upgrade overhead

Reply:
for sure! for what it's worth, this was originally the case, but we changed it to introduce the methods to `optimismPortal`